src/binaryen-c.cpp

```diff
@@ -1030,4 +1072,15 @@ void BinaryenSetAPITracing(int on) {
   }
 }
 
+//
+// ===== Emscripten support =====
+//
+
+#ifdef EMSCRIPTEN
+int __cxa_thread_atexit(void(*)(char*), char*, char*)
```

Comment:
I wasn't able to pinpoint it unfortunately. For what it's worth, I previously tried setting `-s USE_PTHREADS=0` and `-s NO_EXIT_RUNTIME=1`, but compilation still resulted in the same issue, if that helps and if these flags are connected to the issue at all